pkg/controller/controller.go

```diff
@@ -512,6 +514,42 @@ func unmarshalParameters(in []byte) (map[string]interface{}, error) {
 	return parameters, nil
 }
 
+// fetchSecretParameters requests and returns the contents of the given secret as a map
+func fetchSecretParameters(kubeClient kubernetes.Interface, namespace string, secretRef *v1.LocalObjectReference) (map[string]interface{}, error) {
+	if kubeClient == nil {
+		return nil, errors.New("kubeClient is not defined")
+	}
+	secret, err := kubeClient.CoreV1().Secrets(namespace).Get(secretRef.Name, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	result := make(map[string]interface{})
```

Comment:
@duglin yep, it's the same (since we don't explicitly pass extra capacity in the `make` call).
I think it's more about the convention and patterns of usage, see https://groups.google.com/forum/#!topic/golang-nuts/YnqSi2yURSY
```
Similar to new(Struct) vs Struct{}, I use them in slightly different ways:
1) If the value is done and I don't need to add anything to it before releasing it off into the world, I use the {} version
2) If I'm planning on adding things to it (usually programmatically) I use the make() (or new()) version.
```
Since I plan to fill this map below, I'm using `make` there.